src/assign.c

```diff
@@ -288,10 +288,11 @@ int checkOverAlloc(SEXP x)
 }
 
 SEXP alloccolwrapper(SEXP dt, SEXP overAllocArg, SEXP verbose) {
-  if (!IS_TRUE_OR_FALSE(verbose))
-    error(_("%s must be TRUE or FALSE"), "verbose");
+  if ((!isLogical(verbose) && !isInteger(verbose)) || LENGTH(verbose)!=1 || INTEGER(verbose)[0]==NA_INTEGER)
```

Comment:
This if condition is too terse to be readable and relies on the undocumented behaviour of `INTEGER()` on logical input. 

With integer verbose values, we should document the integer codes.

Reply:
We have the same condition here:
https://github.com/Rdatatable/data.table/blob/ff900d1e6a8bcfaa0385bd1304af7b90657d3c4d/src/init.c#L319-L321

Reply:
yeah this if condition is already being used in many parts of the code base where verbose can be both `INTEGER` and `Logical`

Reply:
that tells me we need a helper function :)

Reply:
Same thoughts!